src/Compilers/Core/SharedCollections/ArrayBuilder.cs

```diff
@@ -383,6 +383,14 @@ public void AddRange(ImmutableArray<T> items, int length)
             _builder.AddRange(items, length);
         }
 
+        public void AddRange(T[] items, int start, int length)
+        {
+            for (int i = 0; i < length; i++)
+            {
+                Add(items[start + i]);
```

Comment:
Consider calculating start index and stop index before the loop in order to avoid calculating ```start+i``` on each iteration.